Source/Evolve/WarpXEvolve.cpp

```diff
@@ -205,7 +205,7 @@ WarpX::Evolve (int numsteps)
         // If is_synchronized we need to shift j too so that next step we can evolve E by dt/2.
         // We might need to move j because we are going to make a plotfile.
 
-        int num_moved = MoveWindow(move_j);
+        int num_moved = MoveWindow(step+1, move_j);
```

Comment:
What is the reason that `step+1` is used here? Since the `moving_window_active` routine is checking for `start <= step < end`, it seems that this should be only `step`. If `step+1` is used for example, the value `0` will never be passed into `moving_window_active`.

Reply:
I think originally I just used this argument because in a close-by location I saw `sort_intervals.contains(step+1)`, but that does not mean I understood this well here.

Reply:
Yes, this `step + 1` is a recurring theme in `WarpX::Evolve`. In the end it comes from the fact that there is a shift of 1 between the value of the variable `step` in this routine and the value of the step printed out to the user. I.e., when `step = 0`, we print out `STEP 1 STARTS ...` in the output file.

Here, we should have made it so that the value that the user puts in the input file is consistent with the value that the user sees in output, hence the `+ 1`. I.e., if the user adds `warpx.start_moving_window_step = 100`, he should see something like
```
STEP 100 starts ...
Starting moving window
STEP 100 ends.
```
in the output file. It's true that the value 0 will never be passed, but that's because we only call this routine after `STEP 1 starts` has already been written to the output file.

I'm 100% in favor of harmonizing the values of `step` in the loop and the value printed out to the user (for instance by starting the main loop with `istep[0] = 1`) since this would simplify code readability and reviewing. I can take care of this if needed.

Reply:
I think the difference here is whether `step` describes the current state or it describes the action. For the former, at the start of the calculation, it is at step 0, and then after the particles and fields are advance, it is at step 1. With the latter, the step is the process of doing the advance, so after the code is initialized, it does the first step, i.e. step 1 (there is no step 0). I have thought in terms of the former, but it WarpX seems to be using the latter. In that case, `step+1` is correct.

Reply:
Discussed today: to be simplified in an independent PR by starting from `1` :)